### PR TITLE
- Fix issue #866 and some other small bugs found during implementation.

### PR DESCRIPTION
**Pull request by xiaobenbenecho**
- Fix bug in function plot_series in utils.plotting to plot for both DataFrame and Series. 
- Fix issue  #866, in _get_fre in utils.datetime to extract variant frequency of the series, e.g. 'w-sun'. 
- Fix bug in check_series in utils.validation.series to force all y to series if univariate. 
- Fix bug in _fbprophet and _statsmodels in forecasting.base.adpters to force a univariate data frame to series.

### CODE REVIEW

File: sktime/forecasting/base/adapters/_fbprophet.py [hunk 1]
```diff
@@ -35,6 +36,7 @@ def fit(self, y, X=None, fh=None, **fit_params):
         self._instantiate_model()
         self._check_changepoints()
         self._set_y_X(y, X, enforce_index_type=pd.DatetimeIndex)
+        y, X = check_y_X(y, X)
```
Comment (mloning) [inline sktime/forecasting/base/adapters/_fbprophet.py:39]: This happens inside `self._set_y_X(...)` so not necessary here or any of the other files where this is added in this PR.

File: sktime/utils/datetime.py [hunk 2]
```diff
@@ -53,7 +53,8 @@ def _coerce_duration_to_int(duration, freq=None):
 def _get_freq(x):
     """Get unit for conversion of time deltas to integers"""
     if hasattr(x, "freqstr"):
-        return x.freqstr
+        #e.g. when freq = 'W-SUN'
+        return x.freqstr[0]
```
Comment (mloning) [inline sktime/utils/datetime.py:57]: So this change would simply ignore the `-SUN` information, is my understanding correct? If so, is that a good idea? In either case, we need some testing to make sure this works as expected.

**Review by mloning — COMMENTED** [review]
Thanks for the PR @xiaobenbenecho - but could we limit this to the fix of the frequency handling? Please see my comments below.

**Comment by mloning** [comment]
Hi @xiaobenbenecho I'd love to get the frequency fix merged, let me know if I can help with anything.

**Comment by xiaobenbenecho** [comment]
Hi thanks.
Should I just use the change I have made or you prefer more testing code
regarding the the frequency fix?
Bests

Markus Löning ***@***.***> 于2021年6月12日周六 下午5:52写道：

**Comment by mloning** [comment]
Hi @xiaobenbenecho - let's make this PR a bit more focused, there are currently main files being changed. I suggest to only focus on the frequency fix. Yes, a unit test would be good too.

**Comment by xiaobenbenecho** [comment]
Hi thanks.

Submitted with corresponding test file.

Best regards

**Comment by xiaobenbenecho** [comment]
Hi, yes please. A new PR has been used to fix this issue. Best.

**Pull request closed without merging**